### ISSUE TITLE
gnt-instance migrate should check instance type first

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#756]
Originally reported of Google Code with ID 700.

```
What software version are you running? Please provide the output of "gnt-
cluster --version", "gnt-cluster version", and "hspace --version".

root@vserver8:~# gnt-cluster --version
gnt-cluster (ganeti v2.9.2) 2.9.2

root@vserver8:~# gnt-cluster version
Software version: 2.9.2
Internode protocol: 2090000
Configuration format: 2090000
OS api version: 20
Export interface: 0
VCS version: v2.9.2

<b>What distribution are you using?</b>

ubuntu 12.04.4 LTS


<b>What steps will reproduce the problem?</b>

root@vserver8:~# gnt-instance migrate  instance1
Instance instance1 will be migrated. Note that migration might impact
the instance if anything goes wrong (e.g. due to bugs in the
hypervisor). Continue?
y/[n]/?: y
Failure: prerequisites not met for this operation:
error type: wrong_state, error details:
Instance's disk layout 'plain' does not allow migrations


<b>What is the expected output? What do you see instead?</b>

it is just some cosmetic change but IMHO the prerequisites should be checked before asking the user anything!?

```
Originally added on 2014-01-30 08:40:40 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I agree that this would be a desirable feature, but it is kind of hard with the current architecture. User interaction, like asking questions, are done by the command-line client (i.e., the gnt-instance executable). So they have to happen before the request is handed over to the Ganeti Daemons. However, only there is the knowledge about the configuration, including instance type etc.
```
Originally added on 2014-01-30 09:23:38 +0000 UTC.

Added Labels: Type-Enhancement Priority-Low
Changed State: Accepted
Added to Milestone: Unplanned

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2015-10-07 13:34:11 +0000 UTC.

Changed State: WontFix
